snuba/query/snql/parser.py

```diff
@@ -734,6 +735,44 @@ def parse_snql_query_initial(
     return parsed
 
 
+def _qualify_columns(query: Union[CompositeQuery[QueryEntity], LogicalQuery]) -> None:
+    """
+    All columns in a join query should be qualified with the entity alias, e.g. e.event_id
+    Take those aliases and put them in the table name. This has to be done in a post
+    process since we need to have all the aliases from the join clause.
+    """
+
+    from_clause = query.get_from_clause()
+    if isinstance(query, LogicalQuery) or not isinstance(from_clause, JoinClause):
+        print("REJECTED")
+        return  # We don't qualify columns that have a single source
+
+    # mypy makes this difficult to do non-recursively
+    def find_aliases(aliases: Set[str], join_clause: JoinClause[QueryEntity]) -> None:
+        aliases.add(join_clause.right_node.alias)
+        if isinstance(join_clause.left_node, JoinClause):
+            find_aliases(aliases, join_clause.left_node)
+        elif isinstance(join_clause.left_node, IndividualNode):
+            aliases.add(join_clause.left_node.alias)
```

Comment:
This is already implemented:
https://github.com/getsentry/snuba/blob/master/snuba/query/data_source/join.py#L70